[Bug] Explicitly implemented interface properties cause compile-time error

When explicitly implementing an interface property, Introspection generates Metadata including getter and/or setter which cause compile-time errors because these properties don't belong to the class but to the interface and are, therefore, only accessible when the object is cast to the Interface and not the class.

See [minimal reproduction project](https://github.com/ByteSquire/Introspection-InterfacePropBug)

Example (It doesn't have to be two interfaces, but this is a case where it may be absolutely necessary to use explicit implementation):
```cs
[Meta]
public partial class Class1 : ITest1, ITest2
{
    int ITest1.Hello => 1;
    int ITest2.Hello => 1;
}

public interface ITest1
{
    public int Hello { get; }
}

public interface ITest2
{
    public int Hello { get; }
}
```

The generated code causing compile-time error(s):
```cs
[System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverage]
    public System.Collections.Generic.IReadOnlyList<Chickensoft.Introspection.PropertyMetadata> Properties { get; } = new System.Collections.Generic.List<Chickensoft.Introspection.PropertyMetadata>() {
      new Chickensoft.Introspection.PropertyMetadata(
        Name: "Hello",
        IsInit: false,
        IsRequired: false,
        HasDefaultValue: false,
        Getter: static (object obj) => ((Class1)obj).Hello, <-- compile-time error 
        Setter: null,
        TypeNode: new Chickensoft.Introspection.TypeNode(
          OpenType: typeof(int),
          ClosedType: typeof(int),
          IsNullable: false,
          Arguments: System.Array.Empty<TypeNode>(),
          GenericTypeGetter: static receiver => receiver.Receive<int>(),
          GenericTypeGetter2: default
        ),
        Attributes: new System.Collections.Generic.Dictionary<System.Type, System.Attribute[]>() {
        }
      ), 
      new Chickensoft.Introspection.PropertyMetadata(
        Name: "Hello",
        IsInit: false,
        IsRequired: false,
        HasDefaultValue: false,
        Getter: static (object obj) => ((Class1)obj).Hello, <-- compile-time error
        Setter: null,
        TypeNode: new Chickensoft.Introspection.TypeNode(
          OpenType: typeof(int),
          ClosedType: typeof(int),
          IsNullable: false,
          Arguments: System.Array.Empty<TypeNode>(),
          GenericTypeGetter: static receiver => receiver.Receive<int>(),
          GenericTypeGetter2: default
        ),
        Attributes: new System.Collections.Generic.Dictionary<System.Type, System.Attribute[]>() {
        }
      )
    };
```

Errors:
````
    Introspection-InterfacePropBug/obj/Debug/net8.0/Chickensoft.Introspection.Generator/Chickensoft.Introspection.Generator.TypeGenerator/n_InterfacePropBug_Class129FA4E3125.g.cs(27,54): error CS1061: 'Class1' does not contain a definition for 'Hello' and no accessible extension method 'Hello' accepting a first argument of type 'Class1' could be found (are you missing a using directive or an assembly reference?)
    Introspection-InterfacePropBug/obj/Debug/net8.0/Chickensoft.Introspection.Generator/Chickensoft.Introspection.Generator.TypeGenerator/n_InterfacePropBug_Class129FA4E3125.g.cs(45,54): error CS1061: 'Class1' does not contain a definition for 'Hello' and no accessible extension method 'Hello' accepting a first argument of type 'Class1' could be found (are you missing a using directive or an assembly reference?)
````

--- a/Chickensoft.Introspection.Generator.Tests/test/src/models/DeclaredPropertyTest.cs
+++ b/Chickensoft.Introspection.Generator.Tests/test/src/models/DeclaredPropertyTest.cs
@@ -16,6 +16,7 @@ public class DeclaredPropertyTest {
       IsRequired: false,
       IsNullable: false,
       DefaultValueExpression: null,
+      ExplicitInterfaceName: null,
       TypeNode: new TypeNode(
         "System.String", false, Children: ImmutableArray<TypeNode>.Empty
       ),
@@ -35,6 +36,7 @@ public class DeclaredPropertyTest {
         IsRequired: false,
         IsNullable: false,
         DefaultValueExpression: null,
+        ExplicitInterfaceName: null,
         TypeNode: new TypeNode(
           "System.String", false, Children: ImmutableArray<TypeNode>.Empty
         ),
@@ -50,6 +52,7 @@ public class DeclaredPropertyTest {
       IsRequired: false,
       IsNullable: false,
       DefaultValueExpression: null,
+      ExplicitInterfaceName: null,
       TypeNode: new TypeNode(
         "System.String", false, Children: ImmutableArray<TypeNode>.Empty
       ),
@@ -65,6 +68,7 @@ public class DeclaredPropertyTest {
         IsRequired: false,
         IsNullable: false,
         DefaultValueExpression: null,
+        ExplicitInterfaceName: null,
         TypeNode: new TypeNode(
           "System.String", false, Children: ImmutableArray<TypeNode>.Empty
         ),

--- a/Chickensoft.Introspection.Generator.Tests/test_cases/ExplicitInterfaceProperty.cs
+++ b/Chickensoft.Introspection.Generator.Tests/test_cases/ExplicitInterfaceProperty.cs
@@ -1,0 +1,15 @@
+namespace Chickensoft.Introspection.Generator.Tests.TestCases;
+
+[Meta]
+public partial class ExplicitInterfaceProperty : IInterfaceOne, IInterfaceTwo {
+  string IInterfaceOne.Name => "IInterfaceOne.Name";
+  string IInterfaceTwo.Name => "IInterfaceTwo.Name";
+}
+
+public interface IInterfaceOne {
+  string Name { get; }
+}
+
+public interface IInterfaceTwo {
+  string Name { get; }
+}

--- a/Chickensoft.Introspection.Generator/src/TypeGenerator.cs
+++ b/Chickensoft.Introspection.Generator/src/TypeGenerator.cs
@@ -554,6 +554,7 @@ public class TypeGenerator : IIncrementalGenerator {
           IsInit: isInit,
           IsRequired: isRequired,
           IsNullable: isNullable,
+          ExplicitInterfaceName: property.ExplicitInterfaceSpecifier?.Name.ToString(),
           DefaultValueExpression: defaultValueExpression,
           TypeNode: genericType,
           Attributes: propertyAttributes

--- a/Chickensoft.Introspection.Generator/src/models/DeclaredProperty.cs
+++ b/Chickensoft.Introspection.Generator/src/models/DeclaredProperty.cs
@@ -24,6 +24,7 @@ public sealed record DeclaredProperty(
   bool IsRequired,
   bool IsNullable,
   string? DefaultValueExpression,
+  string? ExplicitInterfaceName,
   TypeNode TypeNode,
   ImmutableArray<DeclaredAttribute> Attributes
 ) {
@@ -32,21 +33,16 @@ public sealed record DeclaredProperty(
     writer.Indent++;
 
     var propertyValue = "value" + (IsNullable ? "" : "!");
+    var typeName = ExplicitInterfaceName ?? typeSimpleNameClosed;
 
     var getter = HasGetter
-      ? $"static (object obj) => (({typeSimpleNameClosed})obj).{Name}"
+      ? $"static (object obj) => (({typeName})obj).{Name}"
       : "null";
 
     var type = TypeNode.ClosedType;
 
-    var setter = HasSetter
-      ? (
-        IsInit
-          ? "null"
-          : $"static (object obj, object? value) => " +
-            $"(({typeSimpleNameClosed})obj)" +
-            $".{Name} = ({type}){propertyValue}"
-      )
+    var setter = HasSetter && !IsInit
+      ? $"static (object obj, object? value) => (({typeName})obj).{Name} = ({type}){propertyValue}"
       : "null";
 
     writer.WriteLine($"Name: \"{Name}\",");

